test/api_api_tokens_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+Unleash API
+
+Testing APITokensAPIService
+
+*/
+
+// Code generated by OpenAPI Generator (https://openapi-generator.tech);
+
+package client
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/Unleash/unleash-server-api-go/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_client_APITokensAPIService(t *testing.T) {
+
+	apiClient := testClient()
+	var createdTokens []string
+
+	t.Run("Test APITokensAPIService CreateApiToken (admin)", func(t *testing.T) {
+
+		expireDate, err := time.Parse(time.RFC3339, "2021-01-01T00:00:00.000Z")
+		require.Nil(t, err)
+		token := client.CreateApiTokenSchemaOneOf{
+			Type:      "admin",
+			TokenName: "test admin token",
+			ExpiresAt: &expireDate,
+		}
+		createTokenSchema := client.CreateApiTokenSchemaOneOfAsCreateApiTokenSchema(&token)
+		resp, httpRes, err := apiClient.APITokensAPI.CreateApiToken(context.Background()).CreateApiTokenSchema(createTokenSchema).Execute()
+
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		assert.Equal(t, 201, httpRes.StatusCode)
+
+		createdTokens = append(createdTokens, resp.Secret)
+	})
+
+	t.Run("Test APITokensAPIService CreateApiToken (client)", func(t *testing.T) {
+
+		expireDate, err := time.Parse(time.RFC3339, "2021-01-01T00:00:00.000Z")
+		require.Nil(t, err)
+		environment := "development"
+		token := client.CreateApiTokenSchemaOneOf2{
+			Type:        "client",
+			TokenName:   "test client token",
+			ExpiresAt:   &expireDate,
+			Environment: &environment,
+		}
+		createTokenSchema := client.CreateApiTokenSchemaOneOf2AsCreateApiTokenSchema(&token)
+		resp, httpRes, err := apiClient.APITokensAPI.CreateApiToken(context.Background()).CreateApiTokenSchema(createTokenSchema).Execute()
+
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		assert.Equal(t, 201, httpRes.StatusCode)
+
+		createdTokens = append(createdTokens, resp.Secret)
+	})
+
+	t.Run("Test APITokensAPIService CreateApiToken (frontend)", func(t *testing.T) {
+
+		expireDate, err := time.Parse(time.RFC3339, "2021-01-01T00:00:00.000Z")
+		require.Nil(t, err)
+		environment := "development"
+		project := "default"
+		token := client.CreateApiTokenSchemaOneOf2{
+			Type:        "frontend",
+			TokenName:   "test frontend token",
+			ExpiresAt:   &expireDate,
+			Environment: &environment,
+			Project:     &project,
+		}
+		createTokenSchema := client.CreateApiTokenSchemaOneOf2AsCreateApiTokenSchema(&token)
+		resp, httpRes, err := apiClient.APITokensAPI.CreateApiToken(context.Background()).CreateApiTokenSchema(createTokenSchema).Execute()
+
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		assert.Equal(t, 201, httpRes.StatusCode)
+
+		createdTokens = append(createdTokens, resp.Secret)
+	})
+
+	t.Run("Test APITokensAPIService DeleteApiToken", func(t *testing.T) {
```

Comment:
Augogenned test?

This is probably good enough for now but it might be nice to have these tests independent of each other's state. Right now this will fail if the others fail, even though delete may still be healthy

Reply:
Well, yes, it'd require us to create a token with the assumption that the API to create a token will work. I think that's better than tests depending on other tests' state

Reply:
In [e599b84](https://github.com/Unleash/unleash-server-api-go/pull/12/commits/e599b84279bf4d6d0ddd0b2b120d4fa925b81bad) I'm creating a token and deleting it afterwards